yt/utilities/lib/particle_kdtree_tools.pyx

```diff
@@ -59,20 +78,25 @@ def generate_smoothing_length(np.float64_t[:, ::1] input_positions,
     cdef uint64_t idx
     cdef uint32_t skipid
     cdef int n_particles = input_positions.shape[0]
-    cdef np.float64_t[:] smoothing_length = np.empty(n_particles)
     cdef np.float64_t tpos, ma, sq_dist
     cdef np.float64_t* pos
+    cdef np.float64_t[:] smoothing_length = np.empty(n_particles)
     cdef uint64_t neighbor_id
     cdef int i, j, k, l, skip
     cdef BoundedPriorityQueue queue = BoundedPriorityQueue(n_neighbors)
+    cdef np.int64_t skipaxis = -1
 
+    cdef axes_range axes
+    set_axes_range(&axes, -1)
+
```

Comment:
this basically just tells the neighbor finding to skip the axis denoted -1, which doesn't exist so we loop over x, y and z (0,1,2)